api/src/main/java/org/openmrs/ChemoDosingInstructions.java

```diff
@@ -0,0 +1,191 @@
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+
+/**
+ * @since 2.4.0
+ */
+public class ChemoDosingInstructions implements DosingInstructions {
+
+	private Double dosageAdjustmentPercentage;
+
+	private String dosageDeliveredTiming;
+
+	private String dosageDilutionInstructions;
+
+	private static final String ADJUSTMENT_PERCENTAGE = "Dosage Adjustment Percentage";
+
+	private static final String DELIVERED_TIMING = "Dosage Delivered Timing";
+
+	private static final String DILUTION_INSTRUCTIONS = "Dosage Dilution Instructions";
+
+	/**
+	 * @see DosingInstructions#getDosingInstructionsAsString(java.util.Locale)
+	 */
+	@Override
+	public String getDosingInstructionsAsString(Locale locale) {
+		StringBuilder dosingInstructions = new StringBuilder();
+
+		// serializing adjustment percentage, timing and dilution instructions to
+		// compose dosing instructions
+		if (this.dosageAdjustmentPercentage != null) {
+			dosingInstructions.append(
+					this.formatDosingInformation(ADJUSTMENT_PERCENTAGE, this.dosageAdjustmentPercentage.toString()));
+		}
+		if (this.dosageDeliveredTiming != null) {
+			dosingInstructions.append(this.formatDosingInformation(DELIVERED_TIMING, this.dosageDeliveredTiming));
+		}
+		if (this.dosageDilutionInstructions != null) {
+			dosingInstructions
+					.append(this.formatDosingInformation(DILUTION_INSTRUCTIONS, this.dosageDilutionInstructions));
+		}
+		return dosingInstructions.toString();
+	}
+
+	/**
+	 * @see DosingInstructions#setDosingInstructions(DrugOrder)
+	 */
+	@Override
+	public void setDosingInstructions(DrugOrder order) {
+
+		// storing dosing instruction type as "ChemoDosingInstructions"
+		order.setDosingType(this.getClass());
+
+		// storing dosing instruction string into DrugOrder object
+		order.setDosingInstructions(this.getDosingInstructionsAsString(Locale.getDefault()));
+	}
+
+	/**
+	 * @see DosingInstructions#getDosingInstructions(DrugOrder)
+	 */
+	@Override
+	public DosingInstructions getDosingInstructions(DrugOrder order) {
+		if (!order.getDosingType().equals(this.getClass())) {
+			throw new APIException("DrugOrder.error.dosingTypeIsMismatched",
+					new Object[] { this.getClass().getName(), order.getDosingType() });
+		}
+		ChemoDosingInstructions chemoDosingInstructions = new ChemoDosingInstructions();
+
+		// dosing instructions in drug order consist of several lines each having a
+		// different type of instruction
+		StringTokenizer tokenizeDosingInstructions = new StringTokenizer(order.getDosingInstructions(), "\n");
+		while (tokenizeDosingInstructions.hasMoreTokens()) {
+
+			// Every instruction line is of the form "instructionLineType :
+			// instructionLineValue"
+			String instructionLine = tokenizeDosingInstructions.nextToken();
+			int indexOfColon = instructionLine.indexOf(':');
+			String instructionLineType = instructionLine.substring(0, indexOfColon - 1);
```

Comment:
why `indexOfColon - 1`? I assume you want to avoid space at the end. It's better to use `trim()` instead of making hard rules that there will be a space always.